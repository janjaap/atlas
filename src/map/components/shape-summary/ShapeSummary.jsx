import React from 'react';
import PropTypes from 'prop-types';

import './_shape-summary.scss';

const ShapeSummary = ({ shapeDistanceTxt, onClearDrawing }) => (
  <div className="shape-summary">
    <span
      className="shape-summary__label"
      aria-label={`Lijn: ${shapeDistanceTxt}`}
    >Lijn: {shapeDistanceTxt}
    </span>
    <button
      className="shape-summary__button"
      onClick={onClearDrawing}
      title="Lijn verwijderen"
    >
      <span className="shape-summary__button--close" />
    </button>
  </div>
);

ShapeSummary.propTypes = {
  shapeDistanceTxt: PropTypes.string.isRequired,
  onClearDrawing: PropTypes.func.isRequired
};

export default ShapeSummary;
