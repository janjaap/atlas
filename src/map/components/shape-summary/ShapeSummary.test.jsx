import React from 'react';
import { shallow } from 'enzyme';

import ShapeSummary from './ShapeSummary';

describe('ShapeSummary', () => {
  let onClearDrawing;
  beforeEach(() => {
    onClearDrawing = jest.fn();
  });

  it('should trigger toggle drawing on when clicked with zero markers', () => {
    const wrapper = shallow(
      <ShapeSummary
        shapeDistanceTxt="23,45 km"
        onClearDrawing={onClearDrawing}
      />
    );
    wrapper.find('button').at(0).simulate('click');
    expect(onClearDrawing).toHaveBeenCalled();
  });

  it('should render info with 2 markers', () => {
    const wrapper = shallow(
      <ShapeSummary
        shapeMarkers={2}
        shapeDistanceTxt="23,45 km"
        onClearDrawing={onClearDrawing}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
