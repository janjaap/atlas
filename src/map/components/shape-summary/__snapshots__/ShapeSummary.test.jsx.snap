// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShapeSummary should render info with 2 markers 1`] = `
<div
  className="shape-summary"
>
  <span
    aria-label="Lijn: 23,45 km"
    className="shape-summary__label"
  >
    Lijn: 
    23,45 km
  </span>
  <button
    className="shape-summary__button"
    onClick={[MockFunction]}
    title="Lijn verwijderen"
  >
    <span
      className="shape-summary__button--close"
    />
  </button>
</div>
`;
